### DISCUSSION TITLE
Anything useful in `topology-viewer`

### DISCUSSION THREAD

**Discussion by zgover** [data-driven-forms/react-forms#1111] [General]
Curious if there is anything useful, rather clarification on the `[data-driven-form/topology-viewer](https://github.com/data-driven-forms/topology-viewer)` public repo? It seems to be actively maintained, anything we could streamline for consumption?

**Comment by rvsia** [comment]
Hello,

in that repo we store a topology viever used in one of our company projects - it’s basically a 2D force node graph in Patternfly 4 styling. It looks like this: 

![](https://user-images.githubusercontent.com/32869456/94815244-99aea900-03fa-11eb-9640-c0d67b810855.png)

However, it has no connection to DDF and we are no longer actively using it, so I would rather avoid using it. But if you find it useful, we are able to provide help/merge PRs.

**Comment by Hyperkid123** [comment]
@zgover as Richard said, there is no connection to DDF other than it was created by us, it has a schema and we needed someplace to put it. It is **not** actively maintained because it is **not** actively used anymore.

> anything we could streamline for consumption?

You should be able to install and run it. But it would certainly not call it production-ready. There are performance limitations. It does not behave very well with around 150+ nodes. The API is not also very friendly, and the documentation is lacking (as you have probably noticed). There is still a potential that it will be used at some point, so we have decided to keep the repository for now.

But the implementation was tailored to our needs so It might not suit everybody. If I were you and had some understanding of [d3 and force-directed graphs](https://observablehq.com/@d3/temporal-force-directed-graph), I would implement my own component. At the end of the day, its just a PF4 wrapper around it with a JSON schema.
